### ISSUE TITLE
Add write transactions support

### ISSUE THREAD

**Issue by bep** [asdine/storm#11]
It would be nice to do something like:

```
accountA.Amount -= 100
accountB.Amount += 100
db.Save(accountA)
db.Save(accountB)
```

As part of a transaction.

**Comment by asdine** [comment]
We can use #7 as an example to do something like this:

```go
....
tx := db.Transaction()
accountA.Amount -= 100
accountB.Amount += 100
tx.Save(accountA)
tx.Save(accountB)
tx.Trigger()
```

**Comment by bep** [comment]
I will think a little bit about this, it's not something I really need right now ...

But all the write operations should support grouped transactions

```
find . -name "*.go" | xargs grep "n.s.Bolt.Update"
./delete.go:	return n.s.Bolt.Update(func(tx *bolt.Tx) error {
./init.go:	err = n.s.Bolt.Update(func(tx *bolt.Tx) error {
./remove.go:	return n.s.Bolt.Update(func(tx *bolt.Tx) error {
./save.go:	return n.s.Bolt.Update(func(tx *bolt.Tx) error {
./set.go:	return n.s.Bolt.Update(func(tx *bolt.Tx) error {
```

**Comment by asdine** [comment]
@bep what do you think?

**Comment by asdine** [comment]
Fixed by #26
